src/cli/migrate/nodeVersion/index.ts

```diff
@@ -212,54 +172,54 @@ const writePatchedContents = async ({
   path,
   contents,
   templated,
-  test,
+  regex,
 }: {
   path: string;
   contents: string;
   templated: string;
-  test?: RegExp;
+  regex?: RegExp;
 }) =>
   await fs.promises.writeFile(
     path,
-    test ? contents.replaceAll(test, templated) : templated,
+    regex ? contents.replaceAll(regex, templated) : templated,
   );
 
-const upgrade = async (
-  { nodeVersion, nodeTypesVersion, ECMAScriptVersion }: Versions,
-  dir: string,
-) => {
+const upgrade = async (versions: Versions, dir: string) => {
   await Promise.all(
-    subPatches.map((subPatch) =>
-      runSubPatch(
-        { nodeVersion, nodeTypesVersion, ECMAScriptVersion },
-        dir,
-        subPatch,
-      ),
-    ),
+    subPatches(versions).map((subPatch) => runSubPatch(dir, subPatch)),
```

Comment:
I think we have cases where 1 file is operated on by more than 1 subpatch. As such, I wonder if we should just run this whole thing sequentially to avoid possible race conditions (i.e. two subpatches both reading the _initial_ state of the file, then the second patch saves over the first patch).

I figure the nested subpatches are meant to address this, but we can probably waste a few extra milliseconds here in exchange for not having to worry about getting the nesting perfectly right.

Reply:
Yeah nice, that cuts out some unnecessary complexity and a few milliseconds doesn't make a significant difference to a already fairly inefficient process 😆